web/src/components/auth/utils.js

```diff
@@ -0,0 +1,194 @@
+    let isAuthenticated = false;
+    if (res && res.data && res.data.access) {
+      isAuthenticated = checkAuthenticated(res.data.access);
+    }
+    if (isAuthenticated) {
+      setAccessTokenOnSuccess(res.data.access);
+      loadUser(res.data.access);
+    }
+    return isAuthenticated;
+  } catch (err) {
+    console.error("Logging-in was unsuccessful", err);
+    setTokensOnFailure();
+  }
+};
+
+/**
+ * Registers a user
+ * @param {string} first_name the user's first name
+ * @param {string} last_name the user's last name
+ * @param {string} email the user's email
+ * @param {string} password the user's password
+ * @param {string} re_password the second validation of the user's password against the first password
+ * @returns {User}
+ */
+export const register = async (
+  first_name,
+  last_name,
+  email,
+  password,
+  re_password
+) => {
+  const options = {
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+  const body = JSON.stringify({
+    email,
+    first_name,
+    last_name,
+    password,
+    re_password,
+  });
+
+  try {
+    const res = await axios.post(apiUrlRegister(), body, options);
+    return { success: res.data };
+  } catch (err) {
+    console.error("Registering was unsuccessful", err);
+  }
+};
+
+/**
+ * 
+ * @returns
+ */
+export const verify = async (uid, token) => {
+  const options = {
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+
+  const body = JSON.stringify({ uid, token });
+
+  try {
+    const res = await axios.post(apiUrlVerifyActivation(), body, options);
+    return { success: res.data };
+  } catch (err) {
+    console.error("Verification failed", err);
+  }
+};
+
+/**
+ * Checks user authentication
+ * @param {string} token
+ * @returns {boolean}
+ */
+export const checkAuthenticated = async (token) => {
+  if (token) {
+    const options = {
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+    };
+
+    const body = JSON.stringify({ token });
+
+    try {
+      const res = await axios.post(apiUrlVerifyToken(), body, options);
+      if (res.data && res.data.code !== "token_not_valid") {
+        return true;
+      } else {
+        console.error("Token is invalid", res);
+        return false;
+      }
+    } catch (err) {
+      console.error("Authentication failed", err);
+    }
+  } else {
+    console.error("Authentication failed: no access token");
+  }
+};
+
+/**
+ * Loads a user
+ * @param {string} token
+ * @returns {User}
+ */
+export const loadUser = async (token) => {
+  const options = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `JWT ${token}`,
+      Accept: "application/json",
+    },
+  };
+
+  try {
+    await axios.get(apiUrlLoadUser(), options);
+  } catch (err) {
+    console.error("Loading user was unsuccessful", err);
+  }
+};
+
+/**
+ * Logs-out a user
+ */
+export const logout = () => {
```

Comment:
```suggestion
```

We shouldn't need to worry about a catch block. If the token doesn't exist, then nothing will happen. We should be able to go without this `logout` function because the generic `removeTokens` should provide the same functionality.